lambdas/process-async-cri-credential/src/main/java/uk/gov/di/ipv/core/processasynccricredential/ProcessAsyncCriCredentialHandler.java

```diff
@@ -224,51 +228,47 @@ private void processSuccessAsyncCriResponse(SuccessAsyncCriResponse successAsync
             }
             verifiableCredentialService.persistUserCredentials(vc);
 
-            sendIpvVcConsumedAuditEvent(auditEventUser, vc);
+            sendIpvVcConsumedAuditEvent(auditEventUser, vc, cri);
         }
     }
 
-    private void validateOAuthState(
-            SuccessAsyncCriResponse successAsyncCriResponse, CriResponseItem criResponseItem)
-            throws AsyncVerifiableCredentialException {
-        if (criResponseItem == null) {
-            LOGGER.error(LogHelper.buildLogMessage("No response item found"));
-            throw new AsyncVerifiableCredentialException(UNEXPECTED_ASYNC_VERIFIABLE_CREDENTIAL);
-        }
-        if (criResponseItem.getOauthState() == null
-                || !criResponseItem
-                        .getOauthState()
-                        .equals(successAsyncCriResponse.getOauthState())) {
-            LOGGER.error(
-                    LogHelper.buildLogMessage(
-                            "State mismatch between response item and async response message"));
-            throw new AsyncVerifiableCredentialException(UNEXPECTED_ASYNC_VERIFIABLE_CREDENTIAL);
-        }
+    private Optional<CriResponseItem> getCriResponseItem(String userId, String state) {
+        final List<CriResponseItem> criResponseItems =
+                criResponseService.getCriResponseItemsByUserId(userId);
+        return criResponseItems.stream()
+                .filter(item -> Objects.equals(item.getOauthState(), state))
+                .findFirst();
     }
 
     private void sendIpvVcReceivedAuditEvent(
             AuditEventUser auditEventUser,
             VerifiableCredential verifiableCredential,
+            Cri cri,
             boolean isSuccessful)
             throws UnrecognisedVotException {
-        AuditEvent auditEvent =
-                AuditEvent.createWithoutDeviceInformation(
-                        AuditEventTypes.IPV_F2F_CRI_VC_RECEIVED,
-                        configService.getParameter(ConfigurationVariable.COMPONENT_ID),
-                        auditEventUser,
-                        getExtensionsForAudit(verifiableCredential, isSuccessful));
-        auditService.sendAuditEvent(auditEvent);
+        if (Cri.F2F.equals(cri)) {
```

Comment:
It might be nicer to put this logic in the parent method but it makes it too complex according the SonarCloud

Reply:
I would agree but I think this is ok for now as we do have a follow-up to handle audit events for both CRIs (possibly with new generic events)